### ISSUE TITLE
VPN: IPsec: Connections - pubkey parameter in swanctl.conf not removed under certain conditions

### ISSUE THREAD

**Issue by Monviech** [opnsense/core#6807]
**Important notices**

Before you add a new report, we ask you kindly to acknowledge the following:

- [x] I have read the contributing guide lines at https://github.com/opnsense/core/blob/master/CONTRIBUTING.md
- [x] I am convinced that my issue is new after having checked both open and closed issues at https://github.com/opnsense/core/issues?q=is%3Aissue

**Describe the bug**

There is a small GUI bug in "VPN: IPsec: Connections". Under certain conditions, "pubkeys" parameter isn't removed in "/usr/local/etc/swanctl/swanctl.conf"
 when using the Local Authentication and Remote Authentication in the GUI, and changing from Pubkey back to Pre-Shared Key.

**To Reproduce**

1. Go to 'VPN: IPsec: Key Pairs'
2. Click on 'Add' and create a new RSA Key, for example 2048 bit.
3. Go to 'VPN: IPsec: Connections'
4. Click on 'Add' and create a new Connection
5. In Local and Remote Authentication, choose "Authentication - Public Key" and in "Public Keys" select the RSA keys you created.
6. Populate all necassary other fields and the Children until you can apply the new connection.
7. Check /usr/local/etc/swanctl/swanctl.conf, you can see that it's correct:
```
           local-dba433e7-2245-4790-a738-8286b6278f48 {
            round = 0
            auth = pubkey
            id = aaa
            pubkeys = ce9ecc03-191f-4601-8e2e-8f390b89a459.pem
```
8. Go back into the Connection you just created and change the Local and Remote Authentication from "Public Key" to "Pre-Shared Key".
9. Check /usr/local/etc/swanctl/swanctl.conf, you can see that it's wrong:
```
        local-dba433e7-2245-4790-a738-8286b6278f48 {
            round = 0
            auth = psk
            id = aaa
            pubkeys = ce9ecc03-191f-4601-8e2e-8f390b89a459.pem
```

**Expected behavior**

When changing the "Authentication" Method of "Local Authentication" and "Remote Authentication" from Public Key to Pre-Shared Key, the "pubkeys = ce9ecc03-191f-4601-8e2e-8f390b89a459.pem" parameter should be removed in the /usr/local/etc/swanctl/swanctl.conf.

**Additional context**

I had this problem because of the GUI bug, I posted additional Log files there:
https://forum.opnsense.org/index.php?topic=35658.0

In summary it results in this particular error:
```
<30>1 2023-08-29T14:34:23+02:00 opn01.example.com charon 52418 - [meta sequenceId="138117"] 06[IKE] <7209bd0f-c7f8-467a-9f8a-6c209d9be771|39> authentication of 'site5.example.com' with pre-shared key successful
<30>1 2023-08-29T14:34:23+02:00 opn01.example.com charon 52418 - [meta sequenceId="138118"] 06[CFG] <7209bd0f-c7f8-467a-9f8a-6c209d9be771|39> constraint check failed: peer not authenticated with peer cert 'site5.example.com'
```

**Environment**

OPNsense 23.7.3-amd64
FreeBSD 13.2-RELEASE-p2
OpenSSL 1.1.1v 1 Aug 2023

**Comment by AdSchellevis** [comment]
I had to read this a couple of times, but you're certainly right, the `pubkeys` should be omitted when not using a public key type.

**Comment by AdSchellevis** [comment]
@Monviech this https://github.com/opnsense/core/commit/a702cf9fb3300e125cd7acc8af3813474606e509 should do the trick, can you try an confirm? Thanks in advance.

**Comment by Monviech** [comment]
Confirmed. Applied the patch and I can't reproduce the bug anymore. Thanks you.
